### ISSUE TITLE
cargo oro-doc-x86_64 opens documentation for aarch64 in browser

### ISSUE THREAD

**Issue by old-df38257622** [oro-os/kernel#33]
Running the `cargo oro-doc-x86_64 --open` command given in the [README.md](https://github.com/oro-os/kernel/blob/master/README.md) will build the documentation as expected, and then proceeds to open `<project root>/target/x86_64-unknown-oro/doc/oro_arch_aarch64/index.html`.

**Comment by Qix-** [comment]
Yeah it's a bit annoying. `cargo doc` will document all crates present in the workspace, even those that aren't used for a particular target architecture. In the case of viewing `oro-arch-aarch64` when building for x86_64, nothing will actually show up for that crate since everything is `#[cfg()]`-gated for just aarch64:

https://github.com/oro-os/kernel/blob/f16fcad8603ea2dbb2d8ec773a0cc021096094f7/oro-arch-aarch64/src/lib.rs#L32

If you build for aarch64, the reverse happens - items show up in `oro-arch-aarch64` but not `oro-arch-x86_64` for example.

Even changing that line to exclude the `doc` config flag doesn't change anything. Neither does re-ordering the packages in `Cargo.toml`, or even using `cargo doc --exclude oro-arch-aarch64`. Rustdoc still insists on documenting the non-target architecture crates, and I assume it defaults to the first in the alphabetical list of workspace crates, which would be `oro-arch-aarch64`.

I'm working on automatic documentation generation via CI that will have a frame that lets you select the architecture. Unfortunately I'm not sure how to get docs.rs-like functionality whereby the architecture members have the e.g. `x86_64 only` tag next to them. Something I've been meaning to ask them about but haven't gotten around to it.

I haven't noticed this before though, so thanks for the report. Just another reason to figure out a better documentation system than the built-in `cargo doc`. For now, there's unfortunately not much I can do about it :/

**Comment by Qix-** [comment]
Fixing this now.
